bigbite/image-comparison
# Issue #9 - Allow block to be resized

## Description

[Issue #9](https://github.com/bigbite/image-comparison/issues/9) - The image comparison block's dimensions are defined by the first image uploaded to the block. This change allows the dimensions of the block to be adjusted further to the initial size of the first image added.

## Change Log

- `block.json` new container height and width attributes have been added.
- the settings panel been extended to include the height and width options.
- within the editor, the image comparison block has the ability to alter the height and width of the block itself. Although the width can be adjusted outside of the block's container, the block itself does not overflow the container when rendered on the front end. This follows the WordPress core Image block's approach.
- the front end render has been updated to reflect the height and width changes made in the editor

## Steps to test

- Check out the branch and run `npm run build:prod` to ensure up to date code
- Add an image comparison block to a page
- Notice, when no images are added, the resize handles are not present
- Add an image to the block
- Notice the handles become visible
- After adding a second image, select the block and resize it using the handles on either the bottom, right, or bottom right of the block
- Add a caption to the block
- Select the block and, under the Settings panel, adjust the height and / or width of the block
- Using the up / down arrow keys, the numbers can be incremented by 1. Holding shift and up / down will increase by 10
- Save the changes and view the page on the front end
- If the assigned width is larger than the container, notice the width of the block is locked at maximum width of that container

## Screenshots/Videos

[A video showing the resize container in action](http://bigbite.im/v/xGxems)
[A screenshot showing a width restricted block and caption](http://bigbite.im/i/duJ8UI)

## Checklist:
